Source/r_plane.c

```diff
@@ -423,6 +413,15 @@ static void do_draw_plane(visplane_t *pl)
         }
         else
         {
+          // Make sure the fade-to-color effect doesn't happen too early
+          if (dc_texturemid < SCREENHEIGHT / 2 * FRACUNIT)
+          {
+            fixed_t diff = dc_texturemid - SCREENHEIGHT / 2 * FRACUNIT;
```

Comment:
I believe this can be optimized even further. If you move the `fixed_t diff = ...` line out of the `if ()` block, you could change the check to `if (diff < 0)` which is indeed the same check as in the following line. What do you think?

Reply:
You're right, it's better. Fixed in https://github.com/fabiangreffrath/woof/pull/488/commits/e821ef8e2a96cb915007770540f24b2b22156949